### PR TITLE
flake-parts: add `nix run` interface along with `devenv container` subcommand and improvements

### PR DESCRIPTION
**Pull request by nazarewk**
The state of implementation of `flake-parts` from #360 wasn't really usable in practice:
1. no way to make an image to upload to registry due to nix2container's patched skopeo not being exposed,
2. no way to run the image because nix2container doesn't package an usable image with current invocations,

This PR makes following additions:
1. make clear the package is not a runnable container by appending it with `-spec`,
2. adds `devenv container [options] [container-name]` family of commands for flakes
	1. adds `--podman-run`
	2. adds local registry aliases
	3. allows passing arguments to docker/podman and container run command itself
3. adds `.envrc`
4. adds default certificates to the container
5. allows passing arbitrary configuration to container build 

TODO:
- [ ] update docs
- [ ] write tests

### CODE REVIEW

File: src/devenv.nix [hunk 1]
```diff
@@ -124,20 +126,33 @@ pkgs.writeScriptBin "devenv" ''
...
+        $copyScript "$spec" --registry "$registry" "''${args[@]}" ''${subcommand[--copy-args]}
       fi
 
-      # docker run
       if [[ ''${subcommand[--docker-run]} != false ]]; then
-        $($CUSTOM_NIX/bin/nix $NIX_FLAGS build --print-out-paths --no-link --impure ".#devenv.containers.\"$container\".dockerRun")
+        # shellcheck disable=SC1090
+        source "$(command -v docopts.sh)"
+        $($CUSTOM_NIX/bin/nix $NIX_FLAGS build --print-out-paths --no-link --impure ".#devenv.containers.\"$container\".dockerRun") -- $(docopt_get_values subcommand '<run-args>')
+      if [[ ''${subcommand[--podman-run]} != false ]]; then
```
Comment (j-baker) [inline src/devenv.nix:152]: there needs to be a 'fi' on the preceding line, or the if needs to be an 'elif'.

**Comment by nazarewk** [comment]
I've successfully built and used the devenv container setup using this branch, so it's ready for review.

**Comment by domenkozar** [comment]
Not to maintain the cacertificate list here, can we just use `dockerTools.caCertificates`?

**Comment by domenkozar** [comment]
Seems like CI is pointing to a regression?

**Comment by nazarewk** [comment]
fixed the failing test (I renamed the output, but didn't touch the tests)

**Comment by nazarewk** [comment]
looks like docopts don't build for Mac?

**Comment by nazarewk** [comment]
@domenkozar how do you handle `docops` on Mac? 

from https://github.com/cachix/devenv/actions/runs/5197759077/jobs/9399642874?pr=503#step:6:921
```shell
error: Package ‘docopts-0.6.4-with-no-mangle-double-dash’ in /nix/store/lm6klhgwfz7inymhgf8siwpypjrvdla7-source/pkgs/development/tools/misc/docopts/default.nix:26 is not supported on ‘x86_64-darwin’, refusing to evaluate.
```

**Comment by domenkozar** [comment]
would be really great if someone can rebase this on top of #745

**Comment by domenkozar** [comment]
I've ported a few things from this PR to #1005, but it would be great to get the rest of the changes in.

**Comment by nazarewk** [comment]
Hello, sorry for lack of updates, but I am currently trying to spread Nix at workplace by using https://github.com/divnix/std + Hive (best intro at https://primamateria.github.io/blog/hive/) and those integrate with https://github.com/numtide/devshell instead. Between this work situation and very different ongoing personal project I doubt I will have time to get back into devenv in foreseeable future.

**Comment by domenkozar** [comment]
Hey, fair enough I'll close this then :)

**Pull request closed without merging**